tests/testthat/test-mle-finder.R

```diff
@@ -0,0 +1,13 @@
+test_that("linalg and optim least-sq coincide", {
+  n_obs <- 32; n_pred <- 4
+  data <- simulate_data(n_obs, n_pred, model = 'linear', seed = 1918)
+  design <- data$design; outcome <- data$outcome
+
+  via_linalg_out <- hiper_glm(design, outcome, model = 'linear')
+  via_bfgs_out <- hiper_glm(
+    design, outcome, model = 'linear', method = 'BFGS'
+  )
+  expect_true(are_all_close(
```

Comment:
-5

The are_all_close function needs to be tested per Ex3.3. It should be in it's own file but Im just leaving the comment here.